app/controllers/devise_passwords_controller.rb

```diff
@@ -13,24 +13,8 @@ def create
 
   # PUT /resource/password
   def update
-    self.resource = resource_class.reset_password_by_token(resource_params)
-    yield resource if block_given?
-
-    if resource.errors.empty?
-      resource.unlock_access! if unlockable?(resource)
-      if Devise.sign_in_after_reset_password
-        flash_message = resource.active_for_authentication? ? :updated : :updated_not_active
-        set_flash_message!(:notice, flash_message)
-        resource.after_database_authentication
-        sign_in(resource_name, resource)
-      else
-        set_flash_message!(:notice, :updated_not_active)
-      end
-      SecurityActivity.change_password!(resource, request.remote_ip)
-      respond_with resource, location: after_resetting_password_path_for(resource)
-    else
-      set_minimum_password_length
-      respond_with resource
+    super do |resource|
```

Comment:
So super calls up to the devise `.update` method right?

Bit i'm not clear on is the order of precidence here.
So does this effectively run everything in devise's `.update` method and _then_ tack our method here onto the end?

Reply:
The block passed to `super` is run by the `yield` call: https://github.com/heartcombo/devise/blob/master/app/controllers/devise/passwords_controller.rb#L35